### ISSUE TITLE
Automatic 'Recent' projects tab in Project Manager

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot-proposals#2952]
### Describe the problem or limitation you are having in your project
Project Manager is really only of use for creating new projects because adding projects to the list is slow, clunky, cumbersome and most people rarely use the 'import' option. It actively promotes bad project management.

The workflow for the majority of people is to download or clone a project, then open the .godot project file. So when re-opening that project they open file explorer, go to the folder and double click the .godot file.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Simple: store recently opened projects in the users global godot area and display them as a tab.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
As below, another tab that stores projects as a simple list within the global user area. Whenever a .godot project is opened it is added to the file if not already in the list.

Management of tabs can reuse all the functionality for managing the 'Projects' list, and having separate tabs separates use-cases. Of course, the 'Projects' tab could be used with a 'toggle' option to show/hide recent projects as an alternative.

![image](https://user-images.githubusercontent.com/12863685/124565046-8cb36000-de39-11eb-9c71-274f44a6a674.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?
No

### Is there a reason why this should be core and not an add-on in the asset library?
Project Manager is not fit for purpose, in my opinion and needs a better way to manage projects, like literally every other IDE and game system. As stated above, manually importing everything has more effort than gain.

**Comment by zinnschlag** [comment]
> Project Manager is really only of use for creating new projects because adding projects to the list is slow, clunky, cumbersome and most people rarely use the 'import' or 'scan' option. It actively promotes bad project management.

> The workflow for the majority of people is to download or clone a project, then open the .godot project file. So when re-opening that project they open file explorer, go to the folder and double click the .godot file. Once a number of projects are done this way, opening becomes slow.

You are making a lot of bold claims about how people use/don't use the project manager without backing it up in any way. This is not an argument against your proposal. But I would like to point out that I am regularly doing most of the things you claim people don't do and I am not doing the things you claim people do. And I am quite happy with it. Of course that is also only one data point.

If this feature is implemented then the project manager should remember which tab (Recent, Project, Templates) was open last and restore this state when restarting, because otherwise the new tab (in first position) would make it more annoying to use for people who prefer the existing project manager.

That being said, there is already a sort option for last modified. I assume that is not good enough for you. Maybe one could add just an "last opened" option instead of adding a whole new tab? I guess that is kinda similar to the toggle option you mentioned.

**Comment by chucklepie** [comment]
Perhaps I was being overly harsh, I'll rephrase.

However, which other editor or development ide do you know that doesn't
store the history of projects previously opened and instead makes you add
them manually to a list?

**Comment by pycbouh** [comment]
I think you just misrepresented what you want to improve. PM already tracks which projects were accessed recently. What you are asking for is that projects opened directly from the file system or CLI are indexed in the PM. It seems this is what you want, not a recent tab.

**Comment by chucklepie** [comment]
Same thing really. Open a project in visual studio, eclipse, etc and they
get stored and ready for next time. Opening projects via the godot project
manager is convoluted, it should be a simple thing, not a workflow of
multiple manual steps.

**Comment by zinnschlag** [comment]
@pycbouh There is actually a bit of a difference in regards to "accessed recently". The project manager does "recently modified". I think what @chucklepie wants is recently opened instead. This can be different, particularly in an import situation when you only want to look at the project.

Moreover, I just did a little test. It seems that the project manager only detects modification of the project file and not the whole project. For example if you just modify a script the project does not move up in the recently modified list. This complicates things further because the project file isn't actually changed that often (at least that is what I see with my projects).

I agree though that these are different things: "Sort by recently opened" and "update recently done something, even when done from outside the project manager". Maybe that should be split into two separate proposals.

**Comment by dalexeev** [comment]
See also [the comment](https://github.com/godotengine/godot-proposals/issues/2329#issuecomment-782939183) (10 likes). We can make it so that each tab remembers the sort separately.

![](https://user-images.githubusercontent.com/47700418/124642790-eac36000-de98-11eb-81f4-5d0423397504.png)

**Comment by pycbouh** [comment]
@zinnschlag Yes, PM doesn't allow sorting by recently opened/accessed and recently modified is not the same, but there is already a proposal for that: https://github.com/godotengine/godot-proposals/issues/2669.

The unique thing about this proposal here is that when a project is opened directly from the file system or from a command line it doesn't get registered with the PM. Those are two separate issues.

**Comment by chucklepie** [comment]
Maybe what I was saying was not being interpreted, I'm referring to projects opened outside of the project manager should be put into a 'recently used' tab. Otherwise, every project you open (which I would say is for a lot of people) via file explorer (e.g. downloaded samples) never appears in Project Manager.

So when you start Godot and want to open one of the many projects you've been playing with, the Godot list is empty and you have to then use file explorer (or file explorer via Godot) and then scan or import the folder before you can start editing the project.
